RadarSDK/RadarAddress.m

```diff
@@ -0,0 +1,165 @@
+    NSString *county;
+    NSString *neighborhood;
+    NSString *number;
+    RadarAddressConfidence *confidence;
+
+    // TODO (jsani): verify serializing RadarCoordinate correctly
+    id coordinateObj = addressDict[@"coordinate"];
+    if (coordinateObj && [coordinateObj isKindOfClass:[RadarCoordinate class]]) {
+        coordinate = (RadarCoordinate *)coordinateObj;
+    }
+
+    id formattedAddressObj = addressDict[@"formattedAddress"];
+    if (formattedAddressObj && [formattedAddressObj isKindOfClass:[NSString class]]) {
+        formattedAddress = (NSString *)formattedAddressObj;
+    }
+
+    id countryObj = addressDict[@"country"];
+    if (countryObj && [countryObj isKindOfClass:[NSString class]]) {
+        country = (NSString *)countryObj;
+    }
+
+    id countryCodeObj = addressDict[@"countryCode"];
+    if (countryCodeObj && [countryCodeObj isKindOfClass:[NSString class]]) {
+        countryCode = (NSString *)countryCodeObj;
+    }
+
+    id countryFlagObj = addressDict[@"countryFlag"];
+    if (countryFlagObj && [countryFlagObj isKindOfClass:[NSString class]]) {
+        countryFlag = (NSString *)countryFlagObj;
+    }
+
+    id stateObj = addressDict[@"state"];
+    if (stateObj && [stateObj isKindOfClass:[NSString class]]) {
+        state = (NSString *)stateObj;
+    }
+
+    id stateCodeObj = addressDict[@"stateCode"];
+    if (stateCodeObj && [stateCodeObj isKindOfClass:[NSString class]]) {
+        stateCode = (NSString *)stateCodeObj;
+    }
+
+    id postalCodeObj = addressDict[@"postalCode"];
+    if (postalCodeObj && [postalCodeObj isKindOfClass:[NSString class]]) {
+        postalCode = (NSString *)postalCodeObj;
+    }
+
+    id cityObj = addressDict[@"city"];
+    if (cityObj && [cityObj isKindOfClass:[NSString class]]) {
+        city = (NSString *)cityObj;
+    }
+
+    id boroughObj = addressDict[@"borough"];
+    if (boroughObj && [boroughObj isKindOfClass:[NSString class]]) {
+        borough = (NSString *)boroughObj;
+    }
+
+    id countyObj = addressDict[@"county"];
+    if (countyObj && [countyObj isKindOfClass:[NSString class]]) {
+        county = (NSString *)countyObj;
+    }
+
+    id neighborhoodObj = addressDict[@"neighborhood"];
+    if (neighborhoodObj && [neighborhoodObj isKindOfClass:[NSString class]]) {
+        neighborhood = (NSString *)neighborhoodObj;
+    }
+
+    id addressNumberObj = addressDict[@"number"];
+    if (addressNumberObj && [addressNumberObj isKindOfClass:[NSString class]]) {
+        number = (NSString *)addressNumberObj;
+    }
+
+    // TODO (jsani): how to serialize RadarAddressConfidence?
+
+    if (coordinate && confidence) {
+        return [[RadarAddress alloc] initWithCoordinate:coordinate formattedAddress:formattedAddress country:country countryCode:countryCode countryFlag:countryFlag state:state stateCode:stateCode postalCode:postalCode city:city borough:borough county:county neighborhood:neighborhood number:number confidence:confidence];
+    }
+
+    return nil;
```

Comment:
There seems to be many boilerplate code. kind of error prone too. I would imagine this would happen for all data types syncing between server and client

Two ideas I could think of now are:
1. we could employ some strong scheme for data serialization during networking, like protobuf
2. we could use some local script (like the remodel tool from Facebook) to auto generate these helper methods

Reply:
@pingxia0823 Yeah there's a lot of repetition in the deserialization and networking code. We should def refactor at some point, can tackle separately though

Reply:
yeah, comment 100% makes sense as we mature with more models / services, but I'm not sure it's in scope for the `v3` release. For now, since my PR only adds one new model (and an enum), I decided to stick with existing conventions rather than require a huge refactor

Reply:
Yeah, just some food for thought